### DISCUSSION TITLE
Ready set Consistency details

### DISCUSSION THREAD

**Discussion by maximecaron** [readysettech/readyset#17] [Q&A] [answered]
I see ReadySet as a replacement to having more read-only database replica.

But i’m wondering if the consistency is weaker than sending a read-only query to a Database replica.

Usually even with replica lag, all row changes made by a write transaction will become visible atomically( no dirty read) at the replica. 
Thus a single query to see all the change of a transaction or none of them but never only a subset.

Will ReadySet offer the same guarantee?

**Comment by vkrasnov** [comment, marked as answer]
ReadySet applies the changes from the primary atomically at a transaction boundary, but only on a per table basis. That means that a view into a single table is always consistent, and queries that only address a single table are always consistent. But a query that performs joins or unions may be temporarily inconsistent following a transaction that modifies two or more tables in the join.
